conda/testing/helpers.py

```diff
@@ -310,129 +320,76 @@ def _patch_for_local_exports(name, subdir_data, channel, index):
     subdir_data._mtime = float("inf")
 
 
-# this fixture appears to introduce a test-order dependency if cached
-def get_index_r_1(subdir=context.subdir):
-    with open(join(TEST_DATA_DIR, "index.json")) as fi:
-        packages = json.load(fi)
+def _get_index_r_base(
+    json_filename,
+    channel_name,
+    subdir=context.subdir,
+    add_pip=False,
+):
+    with open(join(TEST_DATA_DIR, json_filename)) as fi:
+        all_packages = json.load(fi)
+
+    packages = {subdir: {}, "noarch": {}}
+    for key, pkg in all_packages.items():
+        if pkg.get("subdir") == "noarch" or pkg.get("noarch"):
+            packages["noarch"][key] = pkg
+        else:
+            packages[subdir][key] = pkg
+
+    subdir_datas = []
+    channels = []
+    for subchannel, subchannel_pkgs in packages.items():
         repodata = {
             "info": {
-                "subdir": subdir,
+                "subdir": subchannel,
                 "arch": context.arch_name,
                 "platform": context.platform,
             },
-            "packages": packages,
+            "packages": subchannel_pkgs,
         }
 
-    channel = Channel("https://conda.anaconda.org/channel-1/%s" % subdir)
-    sd = SubdirData(channel)
-    with env_var(
-        "CONDA_ADD_PIP_AS_PYTHON_DEPENDENCY",
-        "false",
-        stack_callback=conda_tests_ctxt_mgmt_def_pol,
-    ):
-        sd._process_raw_repodata_str(json.dumps(repodata))
-    sd._loaded = True
-    SubdirData._cache_[channel.url(with_credentials=True)] = sd
-
-    index = {prec: prec for prec in sd.iter_records()}
-    add_feature_records_legacy(index)
-    r = Resolve(index, channels=(channel,))
+        channel = Channel(f"https://conda.anaconda.org/{channel_name}/{subchannel}")
+        channels.append(channel)
+        sd = SubdirData(channel)
+        subdir_datas.append(sd)
+        with env_var(
+            "CONDA_ADD_PIP_AS_PYTHON_DEPENDENCY",
+            "true" if add_pip else "false",
+            stack_callback=conda_tests_ctxt_mgmt_def_pol,
+        ):
```

Comment:
@costrouc, I think the `test_sort_without_prep` errors might have been introduced with this change. Check which solver number is being requested in that test and ensure that the `add_pip` flag is passed as necessary.